media-sound/freac/freac-1.1~alpha20180913.recipe

```diff
@@ -0,0 +1,96 @@
+SUMMARY="A free and open source audio converter"
+DESCRIPTION="fre:ac is a free and open source audio converter. It comes with \
+an integrated tag editor and converts between various formats.
+
+Features include:
+
+- Support for MP3, MP4/AAC, Opus, FLAC and other formats
+- Integrated player and tag editor
+- Support for playlists, cue sheets and chapters
+- Fast, multi-threaded conversion engine
+- Ability to convert to multiple formats at once
+- Can verify lossless input and output files
+- Support for freedb/CDDB queries and submits
+- Support for using command line codecs
+- User interface available in 40+ languages
+- Optional command line interface"
+HOMEPAGE="https://freac.org/"
+COPYRIGHT="2001-2018 Robert Kausch"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/enzo1982/freac/releases/download/v${portVersion/\~alpha/-alpha-}/freac-${portVersion/\~alpha/-alpha-}.tar.gz"
+CHECKSUM_SHA256="5ca1f3b8f548815ced94e8df3b0b736d75879816939661ba1f1baa94fd36f168"
+SOURCE_DIR="freac-${portVersion/\~alpha/-alpha-}"
+PATCHES="freac-$portVersion.patchset"
+ADDITIONAL_FILES="freac.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+SECONDARY_ARCHITECTURES="x86"
+fi
+
+PROVIDES="
+	freac$secondaryArchSuffix = $portVersion
+	cmd:freac = $portVersion
```

Comment:
@Begasus 
> Installing it in $appsDir/freac seems to work, I could even switch language :)
symlinked the binaries to $prefix/bin so they could be launched from cli too, icons show up ok, only thing I could not select was the encoder, but that could be because there wasn't anything to encode also

If the runtimes are in `$appsDir` (and not in a subdirectory of it) then you don't even need to symlink it in `$prefix/bin` because `/boot/system/apps` is already in `$PATH` in a standard system.

So maybe we should install `freeac` in `$appsDir` and `freaccmd` in `$prefix/bin`, without any symlink.
```diff
 PROVIDES="
 	freac$secondaryArchSuffix = $portVersion
-	cmd:freac = $portVersion
+	app:freac = $portVersion
 	cmd:freaccmd = $portVersion
 	"
```

Reply:
runtimes are installed (with the changes I made) to $appsDir/freac/bin, I didn't create a symlink from $appsDir/freac/bin/freac to $appsDir/freac as I think it wouldn't serve any purpose, installing them in $prefix/bin probably would break it again, should be checked ...

Reply:
Hmm, you and some contributors already now that I don't think we should create subdirectories in `$appsDir`. I wish we could, at least for this recipe,  install freac in `$appsDir` instead of `$appsDir/freac/bin`.
If installing it in `$appsDir` does not work then it might be possible to fix it by passing some parameters to `make`.

Now, I also know I'm the only contributor to think we should **not** create subdirectories `$appsDir`, so I guess the majority wins.
(If the final recipe installs a subdirectory in `$appsDir` then I'll let you or someone else merge this PR.)

Reply:
> Now, I also know I'm the only contributor to think we should **not** create subdirectories `$appsDir`, so I guess the majority wins.
> (If the final recipe installs a subdirectory in `$appsDir` then I'll let you or someone else merge this PR.)

That seems like a silly rule.  How would you handle all of an apps support files etc?  You want to put everything into $appsdir?  You are bound to run into conflicts rather quickly if you do that.  I'll put them into $appDir/appname and perhaps just put a symlink in the $appsDir if even that.

Reply:
Sorry, I was not clear enough. The layout I have in mind is:
* `$appsDir/freac`
* `$binDir/freaccmd`
* `$dataDir/freac/` for all the data files, with the possibility to create subdirectories in `$dataDir/freac`.
* `$libDir/freac/` for all the code (i.e. "modules" or "plugins") other than `$appsDir/freac` and `$binDir/freaccmd`, if the program creates such files.
This is imho much cleaner than polluting `/system/apps/` with subdirectories.

Reply:
I didn't say we "should" put every thing in $appsDir, I've checked it that way and it was working that way, better solution would be to change the makefile (or options) to set the right path ...

Reply:
Sorry, you are right. I have "over" reacted a bit :) But I knew you were not suggesting the layout. You were just telling that with your "test" layout the app works.

I just wanted to highlight these issues to let @enzo1982 know that we should look for an alternate layout.